File: CannyStill.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

using Emgu.CV;
using Emgu.CV.CvEnum;
using Emgu.CV.Structure;
using Emgu.CV.UI;

///////////////////////////////////////////////////////////////////////////////////////////////////
namespace CS_test_2
{
    ///////////////////////////////////////////////////////////////////////////////////////////////
    public partial class frmMain : Form 
    {
        // member variables ///////////////////////////////////////////////////////////////////////
        Image<Bgr, Byte> imgOriginal;
        Image<Gray, Byte> imgGrayscale;
        Image<Gray, Byte> imgBlurred;
        Image<Gray, Byte> imgCanny;

        // constructor ////////////////////////////////////////////////////////////////////////////
        public frmMain()
        {
            InitializeComponent();
        }

        ///////////////////////////////////////////////////////////////////////////////////////////
        private void btnOpenFile_Click(object sender, EventArgs e)
        {
            DialogResult drChosenFile;

            drChosenFile = ofdOpenFile.ShowDialog();

            if (drChosenFile != System.Windows.Forms.DialogResult.OK || ofdOpenFile.FileName == "")
            {
                lblChosenFile.Text = "file not chosen";
                return;
            }
            
            try
            {
                imgOriginal = new Image<Bgr, byte>(ofdOpenFile.FileName);
            }
            catch (Exception exception)
            {
                lblChosenFile.Text = "unable to open image, error: " + exception.Message;
                return;
            }

            if (imgOriginal == null)
            {
                lblChosenFile.Text = "unable to open image";
                return;
            }

            imgGrayscale = imgOriginal.Convert<Gray, Byte>();
            imgBlurred = imgGrayscale.SmoothGaussian(5);

            double dblCannyThresh = 180.0;
            double dblCannyThreshLinking = 120.0;

            imgCanny = imgBlurred.Canny(dblCannyThresh, dblCannyThreshLinking);

            ibOriginal.Image = imgOriginal;
            ibCanny.Image = imgCanny;
        }
    }
}

File: CannyStill.vb
'frmMain.vb
'
'NOTE: Do NOT copy/paste the entire text of this file into Visual Studio !! It will not work if you do !!
'Follow the video on my YouTube channel to create the project and have Visual Studio write part of the code for you,
'then copy/pase the remaining text as needed

Option Explicit On
Option Strict On

Imports Emgu.CV
Imports Emgu.CV.CvEnum
Imports Emgu.CV.Structure
Imports Emgu.CV.UI

'''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
Public Class frmMain

    ' member variables ''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
    Dim imgOriginal As Image(Of Bgr, Byte)
    Dim imgGrayscale As Image(Of Gray, Byte)
    Dim imgBlurred As Image(Of Gray, Byte)
    Dim imgCanny As Image(Of Gray, Byte)

    Private Sub btnOpenFile_Click(sender As Object, e As EventArgs) Handles btnOpenFile.Click
        Dim drChosenFile As DialogResult

        drChosenFile = ofdOpenFile.ShowDialog()
        lblChosenFile.Text = ofdOpenFile.FileName

        If (drChosenFile <> Windows.Forms.DialogResult.OK Or ofdOpenFile.FileName = "") Then
            lblChosenFile.Text = "file not chosen"
            Return
        End If

        imgOriginal = New Image(Of Bgr, Byte)(ofdOpenFile.FileName)           ' load image

        imgGrayscale = imgOriginal.Convert(Of Gray, Byte)()

        imgBlurred = imgGrayscale.SmoothGaussian(5)

        Dim dblCannyThresh As Double = 180.0
        Dim dblCannyThreshLinking As Double = 120.0

        imgCanny = imgBlurred.Canny(dblCannyThresh, dblCannyThreshLinking)

        ibOriginal.Image = imgOriginal
        ibCanny.Image = imgCanny
    End Sub

End Class
